listenbrainz/webserver/views/metadata_api.py

```diff
@@ -295,6 +297,137 @@ def get_mbid_mapping():
+        original_result = all_results[original_idx]
+
+        result_dict = query_result.dict(exclude={"index", "artist_credit_arg", "recording_name_arg", "release_name_arg"})
+        original_result.update(**result_dict)
+        all_params.pop(original_idx)
+
+    return all_results, all_params
+
+
+@metadata_bp.route("/lookup/", methods=["POST"])
+@crossdomain
+@ratelimit()
+def get_mbid_mapping_post():
+    """
+    This endpoint is the POST version for looking up recording mbids and associated MusicBrainz data. It allows up to
+    max number of items allowed. (:data:`~webserver.views.api.MAX_ITEMS_PER_GET` items)
+
+    A JSON document with a list of dicts each of which contain metadata (artist_name, recording_name and optionally
+    a release name) for the recording to be looked up. The total number of characters in the artist name, recording name
+    and release name for each recording should be less than or equal to :data:`~webserver.views.metadata_api.MAX_MAPPING_QUERY_LENGTH`.
+
+    .. code:: json
+
+        {
+          "recordings": [
+            {
+              "recording_name": "Never Gonna Give You Up",
+              "artist_name": "Rick Astley",
+              "release_name": "Red Hot"
+            },
+            {
+              "recording_name": "Blinding Lights",
+              "artist_name": "The Weeknd"
+            }
+          ]
+        }
+
+    To see what data this endpoint returns, please look at the data above for the GET version. Note that this endpoint
+    does not support metadata and incs parameters.
+
+    :statuscode 200: lookup succeeded, does not indicate whether a match was found or not
+    :statuscode 400: invalid arguments
+    """
+    data = request.json
+    recordings = data.get("recordings", [])
+    if not recordings:
+        raise APIBadRequest("recordings is invalid or not present in body")
+
+    all_params = {}
+    all_results = []
+    for idx, recording in enumerate(recordings):
+        artist_name = recording.get("artist_name")
+        recording_name = recording.get("recording_name")
+        release_name = recording.get("release_name")
+        if not artist_name:
+            raise APIBadRequest(f"Recording {idx} artist_name is invalid or not present")
+        if not recording_name:
+            raise APIBadRequest(f"Recording {idx} recording_name is invalid or not present")
+        if len(artist_name) + len(recording_name) + len(release_name or "") > MAX_MAPPING_QUERY_LENGTH:
+            raise APIBadRequest(f"Recording {idx} total number of characters in artist_name, recording_name and"
+                                f" release_name arguments must be less than {MAX_MAPPING_QUERY_LENGTH}")
+
+        all_params[idx] = recording
+        all_results.append({
+            "artist_name_arg": artist_name,
+            "recording_name_arg": recording_name,
+            "release_name_arg": release_name,
+            "index": idx
+        })
+
+    try:
```

Comment:
The scalability of this really scares me. :(

Reply:
True, we'll have to work out something better eventually.